packages/flutter/lib/src/material/stepper.dart

```diff
@@ -0,0 +1,665 @@
+    }
+    return null;
+  }
+
+  Widget _buildCircle(int index, bool oldState) {
+    return new Container(
+      margin: const EdgeInsets.symmetric(vertical: 8.0),
+      width: 24.0,
+      height: 24.0,
+      child: new AnimatedContainer(
+        curve: Curves.fastOutSlowIn,
+        duration: kThemeAnimationDuration,
+        decoration: new BoxDecoration(
+          backgroundColor: config.steps[index].isActive ?
+              Theme.of(context).primaryColor : Colors.black38,
+          shape: BoxShape.circle
+        ),
+        child: new Center(
+          child: _buildCircleChild(index, oldState && config.steps[index].state == StepState.error)
+        )
+      )
+    );
+  }
+
+  Widget _buildTriangle(int index, bool oldState) {
+    return new Container(
+      margin: const EdgeInsets.symmetric(vertical: 8.0),
+      width: 24.0,
+      height: 24.0,
+      child: new Center(
+        child: new SizedBox(
+          width: 24.0,
+          height: 20.78, // Height of 24dp-long-sided equilateral triangle.
+          child: new CustomPaint(
+            painter: new _TrianglePainter(
+              color: Colors.red[500]
+            ),
+            child: new Align(
+              alignment: const FractionalOffset(0.5, 0.9), // 0.9 looks better than the geometrical 0.66.
+              child: _buildCircleChild(index, oldState && config.steps[index].state != StepState.error)
+            )
+          )
+        )
+      )
+    );
+  }
+
+  Widget _buildIcon(BuildContext context, int index) {
+    if (config.steps[index].state != _oldStates[index]) {
+      return new AnimatedCrossFade(
+        firstChild: _buildCircle(index, true),
+        secondChild: _buildTriangle(index, true),
+        firstCurve: new Interval(0.0, 0.6, curve: Curves.fastOutSlowIn),
+        secondCurve: new Interval(0.4, 1.0, curve: Curves.fastOutSlowIn),
+        sizeCurve: Curves.fastOutSlowIn,
+        crossFadeState: config.steps[index].state == StepState.error ? CrossFadeState.showSecond : CrossFadeState.showFirst,
+        duration: kThemeAnimationDuration,
+      );
+    } else {
+      if (config.steps[index].state != StepState.error)
+        return _buildCircle(index, false);
+      else
+        return _buildTriangle(index, false);
+    }
+  }
+
+  Widget _buildVerticalControls(BuildContext context) {
+    Color cancelColor;
+
+    switch (Theme.of(context).brightness) {
+      case Brightness.light:
+        cancelColor = Colors.black54;
+        break;
+      case Brightness.dark:
+        cancelColor = Colors.white70;
+        break;
+    }
+
+    assert(cancelColor != null);
+
+    return new Container(
+      margin: const EdgeInsets.only(top: 16.0),
+      child: new ConstrainedBox(
+        constraints: const BoxConstraints.tightFor(height: 48.0),
+        child: new Row(
+          children: <Widget>[
+            new FlatButton(
+              onPressed: config.onStepContinue,
+              color: Theme.of(context).primaryColor,
```

Comment:
The color and text color don't look like they'd work in the dark theme. I would have though that steppers would have resembled the time picker vis the dark the theme.